sdk/Sdk.Generators/FunctionMetadataProviderGenerator.Emitter.cs

```diff
@@ -0,0 +1,159 @@
+
+                    indentedTextWriter.WriteLine($"var {functionBindingsListVarName} = new List<string>();");
+                    AddBindingInfo(indentedTextWriter, functionVariableName, functionBindingsListVarName, function.RawBindings);
+                    indentedTextWriter.WriteLine($"var {functionVariableName} = new DefaultFunctionMetadata");
+                    indentedTextWriter.WriteLine("{");
+                    indentedTextWriter.Indent++;
+                    indentedTextWriter.WriteLine("FunctionId = Guid.NewGuid().ToString(),");
+                    indentedTextWriter.WriteLine("Language = \"dotnet-isolated\",");
+                    indentedTextWriter.WriteLine($"Name = \"{function.Name}\",");
+                    indentedTextWriter.WriteLine($"EntryPoint = \"{function.EntryPoint}\",");
+                    indentedTextWriter.WriteLine($"RawBindings = {functionBindingsListVarName},");
+                    indentedTextWriter.WriteLine($"ScriptFile = \"{function.ScriptFile}\"");
+                    indentedTextWriter.Indent--;
+                    indentedTextWriter.WriteLine("};");
+                    indentedTextWriter.WriteLine($"metadataList.Add({functionVariableName});");
+
+                    functionCount++;
+                }
+            }
+
+            private void AddBindingInfo(IndentedTextWriter indentedTextWriter, string functionVarName, string functionBindingsListVarName, IList<IDictionary<string, string>> bindings)
+            {
+                var bindingCount = 0;
+
+                foreach(var binding in bindings)
+                {
+                    var bindingVarName = functionVarName + "binding" + bindingCount.ToString();
+                    indentedTextWriter.WriteLine($"var {bindingVarName} = new {{");
+                    indentedTextWriter.Indent++;
+                    
+                    foreach(var key in binding.Keys)
+                    {
+                        indentedTextWriter.WriteLine($"{key} = {binding[key]},");
+                    }
+
+                    indentedTextWriter.Indent--;
+                    indentedTextWriter.WriteLine("};");
+                    indentedTextWriter.WriteLine($"var {bindingVarName}JSON = JsonSerializer.Serialize({bindingVarName});");
+                    indentedTextWriter.WriteLine($"{functionBindingsListVarName}.Add({bindingVarName}JSON);");
+
+                    bindingCount++;
+                }
+            }
+
+            private void AddRegistrationExtension(IndentedTextWriter indentedTextWriter)
+            {
+                indentedTextWriter.WriteLine("public static class WorkerHostBuilderFunctionMetadataProviderExtension");
+                indentedTextWriter.WriteLine("{");
+                indentedTextWriter.Indent++;
+                indentedTextWriter.WriteLine("public static IHostBuilder ConfigureGeneratedFunctionMetadataProvider(this IHostBuilder builder)");
```

Comment:
I think this was the initial chosen design b/c creating the extension off `IHostBuilder` makes it easier for users to opt-in since they would only have to add one line.

cc: @fabiocav

Reply:
I'll make a note of this for potential future work - we added it here as the easiest path to registering the provider for a preview version without making significant changes to the core project and just bundled it up in this namespace along with the generated provider. 

To put this generated piece of code in another namespace, I think we'd have to do some design work and refactoring.

Reply:
**Offline convo notes:** If we want to keep to one namespace per file there would be refactoring, but the code can still compile/work with two namespaces in the generated file ... it's not conventionally *best* practice but it does make sense in that everything in this file is still related/connected.